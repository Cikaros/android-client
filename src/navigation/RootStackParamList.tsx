export type RootStackParamList = {
  Home: undefined;
  Onboarding: undefined;
  ChangeServer: undefined;
  LeftDrawer: undefined;
  About: undefined,
  Advance: undefined,
  Peers: undefined,
};
