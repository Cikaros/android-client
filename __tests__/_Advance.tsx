import React from 'react';
import {shallow} from 'enzyme';
import { useSelector} from 'react-redux';
import {RouteProp} from '@react-navigation/native';
import Advance from '../src/screens/Advance';
import {RootStackParamList} from '@navigation/RootStackParamList';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock('@react-navigation/native', () => ({
  useFocusEffect: jest.fn(),
}));

const navigation = {
  navigate: jest.fn(),
  goBack: jest.fn(),
} as any;

const advanceState = {
  inUse: true,
  saving: false,
};
const route = {
  key: 'test-key',
  name: 'Advance',
} as RouteProp<RootStackParamList, 'Advance'>;

const mockStore = configureStore();
const store = mockStore({});

describe('Advance Screen', () => {
  it('should render correctly when pre-shared key is in use', () => {
    const useSelectorMock = useSelector as jest.Mock;
    useSelectorMock.mockReturnValue({...advanceState, inUse: true});
    const wrapper = shallow(
      <Provider store={store}>
        <Advance navigation={navigation} route={route} />
      </Provider>,
    );
    const textComponents = wrapper.find('Text');

    expect(textComponents.at(0).contains('Pre-shared key is in use'));
    expect(
      textComponents
        .at(1)
        .contains('Make sure your other peers have set the same key.'),
    );
    expect(textComponents.at(2).contains('**********'));
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when pre-shared key is not in use', () => {
    const useSelectorMock = useSelector as jest.Mock;
    useSelectorMock.mockReturnValue({...advanceState, inUse: false});

    const wrapper = shallow(
      <Provider store={store}>
        <Advance navigation={navigation} route={route} />
      </Provider>,
    );
    const textComponents = wrapper.find('Text');

    expect(textComponents.at(0).contains('Add a pre-shared key'));
    expect(
      textComponents
        .at(1)
        .contains(
          'You will only communicate with peers that use the same key.',
        ),
    );
    expect(wrapper).toMatchSnapshot();
  });
});
