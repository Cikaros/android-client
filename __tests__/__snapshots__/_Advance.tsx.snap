// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advance Screen should render correctly when pre-shared key is in use 1`] = `
<ContextProvider
  value={
    {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Advance
    navigation={
      {
        "goBack": [MockFunction],
        "navigate": [MockFunction],
      }
    }
    route={
      {
        "key": "test-key",
        "name": "Advance",
      }
    }
  />
</ContextProvider>
`;

exports[`Advance Screen should render correctly when pre-shared key is not in use 1`] = `
<ContextProvider
  value={
    {
      "getServerState": undefined,
      "noopCheck": "once",
      "stabilityCheck": "once",
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Advance
    navigation={
      {
        "goBack": [MockFunction],
        "navigate": [MockFunction],
      }
    }
    route={
      {
        "key": "test-key",
        "name": "Advance",
      }
    }
  />
</ContextProvider>
`;
